akka-actor-typed-tests/src/test/java/akka/actor/typed/ActorSystemTest.java

```diff
@@ -0,0 +1,38 @@
+ /**
+  * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+  */
+package akka.actor.typed;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.scalatest.junit.JUnitSuite;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeoutException;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class ActorSystemTest extends JUnitSuite {
+
+  @Test
+  public void testGetWhenTerminated() throws Exception {
+    final ActorSystem system = ActorSystem.create(Behavior.empty(), "GetWhenTerminatedSystem");
+    system.terminate();
+    final CompletionStage<Terminated> cs = system.getWhenTerminated();
+    cs.toCompletableFuture().get(2, SECONDS);
+  }
+
+  @Test
+  public void testGetWhenTerminatedWithoutTermination() throws Exception {
+    final ActorSystem system = ActorSystem.create(Behavior.empty(), "GetWhenTerminatedWithoutTermination");
+    final CompletionStage<Terminated> cs = system.getWhenTerminated();
+    try {
+      cs.toCompletableFuture().get(2, SECONDS);
```

Comment:
I think this test should rather just look at `isDone` and assert it is false than sleep 2 seconds every time this test is run.

Reply:
What would it test, though? We could just alright remove such test then... I mean, such test succeeds:

```
@Test
public void test() throws Exception {
  system.terminate();
  final CompletionStage<Terminated> cs = system.getWhenTerminated();
  Assert.assertFalse(cs.toCompletableFuture().isDone());
}
```

So, `isDone` doesn't seem to be the right way to check, if system will be terminated soon.

Reply:
Why would it be terminated soon if you didn't terminate it, and if you need to test that it doesn't become terminated on its own why is 2 seconds enough?

I think the only value of the current test is to verify that when the actor system is not terminated, the CS isn't completed either, and that can be tested just fine with `isDone`. 

The other test should still do the blocking wait for termination though, as it verifies that some time soon after terminating the actor system the CS completes.

Reply:
But when the actor system is terminated, the CS isn't completed immediately too, what is shown in the test from my previous comment :)

I suggest to just decrease timeout from 2 seconds to something like 200 millis, if 2 seconds is unacceptable. Then, if some future change will broke this method, test will fail at least sometimes.

Reply:
What you showed in the sample in the comment is a indeterministic test, you can never know that the CS completion has happened before that assertion is reached or not, so for that case it makes sense to wait for a maximum time period, when we know the completion will arrive.

To make the opposite test, that verifies that the CS doesn't magically complete on its own, you would have to wait forever though, which isn't really very useful.

What is potentially useful is to know that the CS returned from a running system isn't already completed, if we poll this during an hour, two seconds, 200 milliseconds or just check it right away doesn't really matter - we wont be more certain that it doesn't magically complete just after our span ends. Therefore `isDone` is perfectly fine for this test.

Reply:
Ok, thanks for detailed explanation. Now that makes sense to me too.